test/sql/upsert/upsert_conflict_in_different_chunk.test

```diff
@@ -11,34 +11,60 @@ statement ok
 SET threads=1;
 
 statement ok
-CREATE TABLE create_or_replace(i BIGINT PRIMARY KEY, s VARCHAR);
+CREATE TABLE create_or_replace(
+	i BIGINT PRIMARY KEY,
+	s VARCHAR
+);
 
 statement ok
-CREATE TABLE inserts (i BIGINT, s VARCHAR);
+CREATE TABLE inserts (
+	i BIGINT,
+	s VARCHAR
+);
 
 statement ok
-INSERT INTO inserts VALUES (1, 'hello'), (1, 'world');
+INSERT INTO inserts VALUES
+	(1, 'hello'),
+	(1, 'world');
 
 # We cannot perform on-conflict handling in the same chunk. This is a known limitation.
-statement maybe
-INSERT OR REPLACE INTO create_or_replace SELECT i, s FROM inserts;
-----
-PRIMARY KEY or UNIQUE constraint violated
+statement ok
+INSERT OR REPLACE INTO create_or_replace
+SELECT
+	i,
+	s
+FROM inserts;
 
 statement ok
 DELETE FROM inserts;
 
 statement ok
-INSERT INTO inserts (SELECT range, 'hello' FROM range(2050) UNION SELECT 1, 'world');
+INSERT INTO inserts (
+	SELECT
+		range,
+		'hello'
+	FROM range(2050) UNION
+	SELECT
+		1,
+		'world'
+	);
 
 # We can perform on-conflict handling in a different chunk. Appending duplicated values with INSERT OR REPLACE in the
 # same statement/transaction is currently not officially supported (known limitations).
 # We fix it for this case, as we introduced changes in https://github.com/duckdb/duckdb/pull/7407 that would
 # otherwise throw an InternalException in the segment tree: https://github.com/duckdb/duckdb/issues/11924.
-statement ok
+
+# This *was* fixed because the previous statement errored (introducing a conflicting tuple in the same Vector)
+# But that is now fixed, so this is now attempting to update the same existing row_id twice, once in the first chunk and once in the second
+# which is still a problem.
+
+# Previously this was inserting a tuple into transaction-local storage and then updating that tuple in the next chunk, which *is* supported.
+statement error
 INSERT OR REPLACE INTO create_or_replace SELECT i, s FROM inserts;
+----
+ON CONFLICT DO UPDATE can not update the same row twice in the same command. Ensure that no rows proposed for insertion within the same command have duplicate constrained values
```

Comment:
Let's copy other systems (postgres) here.

Reply:
Ah, actually explicitly disabled by postgres
```sql
postgres=# INSERT INTO integers(i,j) select i%5,i from generate_series(1, 5000) tbl(i) on conflict (i) do update set j = excluded.j, k = excluded.i;
```
```
ERROR:  ON CONFLICT DO UPDATE command cannot affect row a second time
HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values.
```

Reply:
Good to know it's not a limitation of the system, but merely behavior we're adopting from other systems.

Reply:
Yes, and it makes sense from the SQL semantics - as there is no defined order on the rows.

Reply:
Yea, we could probably add a flag to disable it when we can guarantee insertion order though
- single threaded
- insertion order preservation is on
- not batched